### PR TITLE
Hw_01

### PR DESCRIPTION
**Pull request by vlykhvar**

### CODE REVIEW

File: src/com/company/Main.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.company;
+
+public class Main {
+    public static void main(String[] args) {
+        Counter counter = new Counter();
+        Thread first = new Threads();
+        Thread second = new Thread(new Runnables());
```
Comment (YuraBondarenko) [inline src/com/company/Main.java:7]: make methods in class Counter not static and pass `counter` into constructor

**Review by YuraBondarenko — CHANGES_REQUESTED** [review]

**Review by YuraBondarenko — APPROVED** [review]

File: src/com/company/Counter.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    public Counter() {
+        count = 0;
+    }
+
+    public int getCount() {
+        return count;
+    }
+
+    public void setCount(int count) {
+            this.count += count;
```
Comment (RomanDubovskyi) [inline src/com/company/Counter.java:15]: it looks like regular setter but has unprodictable logic, let's fix it 
(if you want to have method that increments your count value create a separate one)

File: src/com/company/Runnables.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    public final Counter counter;
+
+    public Runnables(Counter counter) {
+        this.counter = counter;
+    }
+
+    @Override
+    public void run() {
+        for (int j = 0; j < 100; j++) {
+            counter.setCount(1);
```
Comment (RomanDubovskyi) [inline src/com/company/Runnables.java:13]: ?

File: src/logging [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+Thread-0: counter is 191
+Thread-0: counter is 192
+Thread-0: counter is 193
+Thread-0: counter is 194
+Thread-0: counter is 195
+Thread-0: counter is 196
+Thread-0: counter is 197
+Thread-0: counter is 198
+Thread-0: counter is 199
+Thread-0: counter is 200
```
Comment (RomanDubovskyi) [inline src/logging:200]: empty line at the end of the file

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/com/company/Runnables.java (hunk 3 above)
Comment (vlykhvar) [reply src/com/company/Runnables.java]: chenged to logic name

File: src/com/company/Counter.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        return count;
+    }
+
+    public void setCount(int count) {
+            this.count = count;
+    }
+
+    public int addOne() {
+        count++;
+        return count;
```
Comment (RomanDubovskyi) [inline src/com/company/Counter.java:20]:
```suggestion
    public int increment() {
         return ++count;
```

File: src/com/company/Threads.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
+package com.company;
+
+import org.apache.log4j.Logger;
+
+public class Threads extends Thread {
+    public final Counter counter;
```
Comment (RomanDubovskyi) [inline src/com/company/Threads.java:6]:
```suggestion
    private final Counter counter;
```
we don't need it to be public

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**